libs/esp32kernel/esp32kernel.erl

```diff
@@ -0,0 +1,303 @@
+%
+% This file is part of AtomVM.
+%
+% Copyright 2023 Davide Bettio <davide@uninstall.it>
+%
+% Licensed under the Apache License, Version 2.0 (the "License");
+% you may not use this file except in compliance with the License.
+% You may obtain a copy of the License at
+%
+%    http://www.apache.org/licenses/LICENSE-2.0
+%
+% Unless required by applicable law or agreed to in writing, software
+% distributed under the License is distributed on an "AS IS" BASIS,
+% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+% See the License for the specific language governing permissions and
+% limitations under the License.
+%
+% SPDX-License-Identifier: Apache-2.0 OR LGPL-2.1-or-later
+%
+
+-module(esp32kernel).
+
+-export([
+    start/0, start_repl/1, start_network/0, handle_req/3, erase_net_config/0, save_net_config/2
+]).
+
+-record(nc_state, {socket, pending_pid, pending_ref}).
+
+-define(DEFAULT_AP_SSID, <<"AtomVM-ESP32">>).
+-define(DEFAULT_AP_PSK, <<"esp32default">>).
+-define(DEFAULT_WEB_SERVER_PORT, 8080).
+
+start() ->
+    erlang:display("AtomVM kernel..."),
+
+    erlang:display("Starting network..."),
+    spawn(?MODULE, start_network, []),
```

Comment:
It's probably best to leave the network initialization and setup to the user. However it would be very nice to have this network setup boilerplate in a different module but ready to be included within any application supervision tree.

Reply:
good point, however the rationale is very simple: developer and debugging experience.
By doing this you have the chance to have a remote shell (ALISP console) and to configure parameters useful for first connection. As a second step I would also like allow uploading the application .avm defeating the mandatory need for a USB cable or for esptool after the first flashing.
Network can be disable anyway using NVS and also NVS can be pre-programmed.

Reply:
Now it is handled as discussed.